src/Compilers/Core/Portable/Compilation/ConversionInfo.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Semantics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Summarizes whether a conversion is allowed, and if so, which kind of conversion (and in 
+    /// some cases, the associated symbol).
+    /// </summary>
+    public struct ConversionInfo : IEquatable<ConversionInfo>
```

Comment:
I've made the common type a struct to prevent unnecessary heap allocations if i were to go with something like an interface.  The representation of the data is as two fields.  One enum member that encodes all the bits of the Conversion, and one pointer to hold the IMethodSymbol if this is a user defined conversion.

This representation closely matches the Conversion types from both VB and C#.  And it attempts to provide a language neutral API for the concepts that those exactly share in common.